modules/collect/src/main/java/com/opengamma/strata/collect/result/FailureItem.java

```diff
@@ -69,6 +72,82 @@
   @PropertyDefinition(get = "optional")
   private final Class<? extends Exception> causeType;
 
+  //-------------------------------------------------------------------------
+  /**
+   * Obtains a failure from a reason and message.
+   * <p>
+   * An exception will be created internally to obtain a stack trace.
+   * The cause type will not be present in the resulting failure.
+   * 
+   * @param reason  the reason
+   * @param message  the failure message, not empty
+   * @return the failure
+   */
+  public static FailureItem of(FailureReason reason, String message) {
```

Comment:
I imagine a common use case would be something like:

    FailureItem item = FailureItem.of(FailureReason.ERROR, Messages.format("Couldn't build a Foo from {}", fooStr);

Would it be worth having that built in?

    public static FailureItem of(FailureReason reason, String message, Object... messageArgs)